datafusion/sqllogictest/test_files/order.slt

```diff
@@ -1260,3 +1260,34 @@ limit 2;
 
 statement ok
 drop table ordered_table;
+
+query TT
+EXPLAIN SELECT
+    CASE
+        WHEN name = 'name1' THEN 0.0
+        WHEN name = 'name2' THEN 0.5
+    END AS a
+FROM (
+    SELECT 'name1' AS name
+    UNION ALL
+    SELECT 'name2'
+)
+ORDER BY a DESC;
```

Comment:
Can you please also add a test case with query that would produce incorrect results before the fix?

Reply:
Sure, I have added a new test in this [commit](https://github.com/apache/datafusion/pull/13497/commits/fa9284a6d8c95073e53a0fb38760f7cf3056ed5d). Prior to this fix, we would get nondeterministic results.